inference-engine/src/auto_plugin/auto_exec_network.cpp

```diff
@@ -13,40 +13,118 @@
 namespace AutoPlugin {
 using namespace InferenceEngine;
 
-AutoExecutableNetwork::AutoExecutableNetwork(const SoExecutableNetworkInternal& network, bool enablePerfCount) :
-    _network(network), _enablePerfCount(enablePerfCount) {
+AutoExecutableNetwork::AutoExecutableNetwork(NetworkFuture cpuFuture,
+                                             NetworkFuture acceleratorFuture,
+                                             bool          enablePerfCount)
+                                             : _cpuFuture(std::move(cpuFuture))
+                                             , _acceleratorFuture(std::move(acceleratorFuture))
+                                             , _enablePerfCount(enablePerfCount) {
+    // both are valid, like AUTO:CPU,GPU
+    if (_cpuFuture.valid() && _acceleratorFuture.valid()) {
+        _networkFirstReady = _cpuFuture.get();
```

Comment:
if CPU loading failed, and the CPU is NOT the accelerator, we should catch the exception (whihc the future::get() will re-throw).
this is captured as " // TODO: FIXME: revert the exception handling logic back to gracefully handle LoadNetwork failures" in another place but should be handled here I guess

Reply:
agreed, addressed, thanks!
BTW, I want to keep debug log until no any other issues since may still need to debug if code change.